### ISSUE TITLE
Full screen optimization of status bar

### ISSUE THREAD

**Issue by Visualizeit** [spckio/spck-issues#111]
![screen](https://user-images.githubusercontent.com/61679509/76380730-88367e80-638e-11ea-8db5-0616455aa309.png)
Status bar has black edge
Immersed transparent status bar

**Comment by spck-io** [comment]
What optimization are you looking for? Just a note, Google guidelines recommend not hiding the status bar for apps unless absolutely necessary.

**Comment by spck-io** [comment]
I have recently experimented with using full screen mode for the app and it caused a lot of issues. I have concluded it isn’t feasible until newer APIs are introduced in Android R

1. Legacy full screen mode: eliminates the ability to resize the editor when soft keyboard is used. This causes a variety of problems mainly with the extra keyboard menu being hidden by the keyboard and various other interface hidden by the keyboard.

2. Android 4.1 full screen mode: used workarounds using custom frames to get around the keyboard resize issue and app resume to get around app exiting full screen from notifications and swipe, but can not find workaround to prevent exit of full screen mode when soft keyboard is active. Status bar cannot be hidden independently of keyboard and causes unwanted artifacts when forcefully hiding the status bar or navigation when keyboard is active. Defeats the whole purpose of full screen can only be used without soft keyboard (where screen estate really matters)
